File: cocmd_cli/src/bin/cmd/add.rs
use anyhow::Result;
use cocmd::core::source::Source;
use cocmd::core::sources_manager::SourcesManager;
use cocmd::utils::repository::find_cocmd_files;
use console::Style;
use dialoguer::Confirm;
use std::path::Path;
use tracing::info;

pub fn add_source(sources_manager: &mut SourcesManager, source: &str) -> Result<cocmd::CmdExit> {
    info!("Add source {:?}", source);

    let source_label = if source == "demo" {
        let resources_path = Path::new(env!("CARGO_MANIFEST_DIR")).join("resources");
        resources_path.join(source)
    } else {
        Path::new(source).to_owned()
    };

    let locations = find_cocmd_files(&source_label, sources_manager.settings.scan_depth);

    let lst_locs = locations.join("\n  - ");
    let style = Style::new().bold().green();
    println!(
        "found {} cocmd sources in this path:\n  - {}",
        locations.len(),
        lst_locs
    );

    if Confirm::new()
        .with_prompt("Do you want to continue?")
        .interact()?
    {
        for loc in locations {
            let source = Source::new(&loc, &sources_manager.settings);
            if let Ok(source_res) = source {
                sources_manager.add_source(source_res);
                println!("{}", style.apply_to(format!("Source '{:?}' added", loc)));
            }
        }
    } else {
        println!("{}", style.apply_to("Skipped. you answered 'NO'"));
    }

    Ok(cocmd::CmdExit {
        code: exitcode::OK,
        message: None,
    })
}

File: cocmd_old/cocmd_cli_py/commands/subcommands/add.py
import click

from cocmd.core.source import Source
from cocmd.utils.repository import find_cocmd_files
from ..groups import add
from cocmd.settings import click_pass_settings
import inquirer
from cocmd.utils.console import console
from cocmd import resources
import os


@add.command()
@click.argument("source")
@click_pass_settings
def source(settings, source: str):
    """
    add a source
    """

    source_label = source

    if source_label == "demo":
        source_label = os.path.join(os.path.dirname(resources.__file__), source_label)

    locations = find_cocmd_files(source_label, settings.scan_depth)

    lst_locs = "\n  - ".join(locations)
    console.print(
        f"""found {len(locations)} cocmd sources in this path:
  - {lst_locs}
    """
    )
    questions = [
        inquirer.Confirm("sure", message="Continue?", default=True),
    ]

    answers = inquirer.prompt(questions)

    if answers["sure"]:
        for loc in locations:
            source = Source(loc, settings)
            settings.sources_manager.add_source(source)
            console.print(f"[bold green]Source '{source}' added")
    else:
        console.print("[bold red]Skipped. you answered 'NO'")
